File: Cocoa Programming/Cocoa Pr L62 ObjC/Cocoa Pr L62 ObjC/ViewController.m
//
//  ViewController.m
//  Cocoa Pr L62 ObjC
//
//  Created by Michele Galvagno on 09/06/22.
//

#import "ViewController.h"

@implementation ViewController

@synthesize textField;

- (void)viewDidLoad {
    [super viewDidLoad];

    // Do any additional setup after loading the view.
}


- (void)setRepresentedObject:(id)representedObject {
    [super setRepresentedObject:representedObject];

    // Update the view, if already loaded.
}

- (void)prepareForSegue:(NSStoryboardSegue *)segue sender:(id)sender {
    if ([[segue identifier]  isEqualToString: @"DetailSegue"]) {
        NSViewController *detailVC = (NSViewController *)segue.destinationController;
        detailVC.title = segue.identifier;
        detailVC.representedObject = textField.stringValue;
    }
}


@end

File: Cocoa Programming/Cocoa Pr L62 Storyboard/Cocoa Pr L62 Storyboard/ViewController.swift
//
//  ViewController.swift
//  Cocoa Pr L62 Storyboard
//
//  Created by Michele Galvagno on 07/06/22.
//

import Cocoa

class ViewController: NSViewController {
    @IBOutlet weak var nameField: NSTextField!
    
    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }

    override var representedObject: Any? {
        didSet {
        // Update the view, if already loaded.
        }
    }
    
    override func prepare(for segue: NSStoryboardSegue, sender: Any?) {
        if segue.identifier == "DetailSegue" {
            (segue.destinationController as! NSViewController).representedObject = nameField.stringValue
        }
    }


}

